src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs

```diff
@@ -117,7 +117,7 @@ public virtual Task InvokeAsync()
             }
             else
             {
-                return Task.FromException(scopeException);
+                return Task.FromException(scopeException!);
```

Comment:
Need for `!` is a compiler bug though a fairly minor one. The conditions above eliminate the `scopeException == null` case but the compiler isn't smart enough to realize it. @jaredpar @jasonmalinowski worth filing a bug?

Reply:
@agocke @jcouv can help answer that one.

Reply:
I think this is worth a bug. It may end up being by design, but it's a useful demonstration of the limits of our analysis.

Reply:
dotnet/roslyn#36535